src/pages/dataElements/fields/AggregationTypeField.tsx

```diff
@@ -51,19 +53,36 @@ export function AggregationTypeField() {
 
     return (
         <FieldRFF
+            // See https://final-form.org/docs/react-final-form/types/FieldProps#validate
+            // for why this is necessary
+            key={disabled ? 0 : 1}
             disabled={disabled}
             component={SingleSelectFieldFF}
             dataTest="dataelementsformfields-aggregationtype"
-            required
+            required={!disabled}
             inputWidth="400px"
             name="aggregationType"
-            label={i18n.t('{{fieldLabel}} (required)', {
-                fieldLabel: i18n.t('Aggregation type'),
-            })}
+            label={
+                disabled
+                    ? i18n.t('Aggregation type')
+                    : i18n.t('{{fieldLabel}} (required)', {
+                          fieldLabel: i18n.t('Aggregation type'),
+                      })
+            }
             helpText={helpText}
             options={options || []}
             validateFields={[]}
-            validate={required}
+            // @TODO: Why can I not use `FormValues` here?
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            validate={(value: string, values: any) => {
+                const { valueType } = values
+                // Using the `disabled` value from above causes an issue:
```

Comment:
Would be nice to figure out if this can be avoided. I guess using `ref` would work, but that's not clean either... I wonder why referencing `disabled` would be any different than referencing `values` from the component-scope...

Reply:
> Ah I guess it's related to the effect that run when disabled changes. And then since the key resets the state you have a different component?

Yes, I think that is the case